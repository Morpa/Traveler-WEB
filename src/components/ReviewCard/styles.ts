import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  display: flex;
  padding: 3rem;
  max-width: 60rem;
`
export const ImageWrapper = styled.div`
  ${({ theme }) => css`
    img {
      margin-top: -1rem;
      margin-right: ${theme.spacings.small};
      width: 6.4rem;
      height: 6.4rem;
      border-radius: 50%;
      object-fit: cover;
    }
  `}
`
export const ContentWrapper = styled.div`
  ${({ theme }) => css`
    h3 {
      color: ${theme.colors.textColor};
      margin-right: 10rem;
    }
  `}
`

export const Header = styled.div`
  display: flex;
  align-items: center;
`
export const Review = styled.p`
  ${({ theme }) => css`
    padding: 1rem 0;
    font-weight: ${theme.font.light};
    font-size: ${theme.font.sizes.medium};
    color: ${theme.colors.textColor};
    line-height: ${theme.spacings.small};
  `}
`
