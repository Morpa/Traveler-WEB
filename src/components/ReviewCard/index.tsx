import { useEffect, useState } from 'react'
import BeautyStars from 'beauty-stars'

import * as S from './styles'

export type ReviewCardProps = {
  name: string
  imageUrl: string
  description: string
  starry: number
}

const ReviewCard = ({
  name,
  imageUrl,
  description,
  starry
}: ReviewCardProps) => {
  useEffect(() => {
    setStars(starry)
  }, [starry])

  const [stars, setStars] = useState(0)
  return (
    <S.Wrapper>
      <S.ImageWrapper>
        <img src={imageUrl} alt={`Avatar of ${name}`} />
      </S.ImageWrapper>

      <S.ContentWrapper>
        <S.Header>
          <h3>{name}</h3>
          <BeautyStars value={stars} onChange={(value) => setStars(value)} />
        </S.Header>

        <S.Review>{description}</S.Review>
      </S.ContentWrapper>
    </S.Wrapper>
  )
}

export default ReviewCard
