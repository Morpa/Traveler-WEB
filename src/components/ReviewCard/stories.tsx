import { Story, Meta } from '@storybook/react/types-6-0'
import ReviewCard, { ReviewCardProps } from '.'

export default {
  title: 'ReviewCard',
  component: ReviewCard,
  args: {
    name: 'Patricksom Vieras',
    imageUrl: 'https://source.unsplash.com/user/willianjusten/300x140',
    description:
      'Todos os produtos comercializados s√£o de excelente qualidade, recomendo!',
    starry: 4
  }
} as Meta

export const Default: Story<ReviewCardProps> = (args) => (
  <ReviewCard {...args} />
)
