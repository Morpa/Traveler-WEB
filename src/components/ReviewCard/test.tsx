import { fireEvent, screen } from '@testing-library/react'
import BeautyStars from 'beauty-stars'

import { renderWithTheme } from 'utils/tests/helpers'

import ReviewCard from '.'

const props = {
  name: 'Patricksom Vieras',
  imageUrl: 'https://source.unsplash.com/user/willianjusten/300x140',
  description:
    'Todos os produtos comercializados s√£o de excelente qualidade, recomendo!',
  starry: 4
}

describe('<ReviewCard />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<ReviewCard {...props} />)

    expect(
      screen.getByRole('heading', { name: props.name })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('img', { name: `Avatar of ${props.name}` })
    ).toHaveAttribute('src', props.imageUrl)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render stars', () => {
    const star = jest.fn()

    renderWithTheme(
      <ReviewCard {...props}>
        <BeautyStars value={4} onChange={star} />
      </ReviewCard>
    )

    fireEvent.click(screen.getByRole('listitem', { name: /5 star/i }))
  })
})
