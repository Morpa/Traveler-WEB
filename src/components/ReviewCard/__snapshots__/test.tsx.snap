// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewCard /> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 3rem;
  max-width: 60rem;
}

.c1 img {
  margin-top: -1rem;
  margin-right: 2.4rem;
  width: 6.4rem;
  height: 6.4rem;
  border-radius: 50%;
  object-fit: cover;
}

.c2 h3 {
  color: #617480;
  margin-right: 10rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  padding: 1rem 0;
  font-weight: 300;
  font-size: 1.6rem;
  color: #617480;
  line-height: 2.4rem;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <img
      alt="Avatar of Patricksom Vieras"
      src="https://source.unsplash.com/user/willianjusten/300x140"
    />
  </div>
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <h3>
        Patricksom Vieras
      </h3>
      <ul
        style="color: rgb(18, 22, 33); margin: 0px; padding: 0px; list-style: none; display: flex;"
      >
        <li
          style="cursor: pointer; position: relative; margin-right: 16px;"
          title="1 star"
        >
          <svg
            fill="#F25D27"
            height="18"
            viewBox="0 0 19 18"
            width="19"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.33333 1L11.9083 6.21667L17.6667 7.05834L13.5 11.1167L14.4833 16.85L9.33333 14.1417L4.18333 16.85L5.16667 11.1167L1 7.05834L6.75833 6.21667L9.33333 1Z"
              stroke="#F25D27"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </li>
        <li
          style="cursor: pointer; position: relative; margin-right: 16px;"
          title="2 star"
        >
          <svg
            fill="#F25D27"
            height="18"
            viewBox="0 0 19 18"
            width="19"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.33333 1L11.9083 6.21667L17.6667 7.05834L13.5 11.1167L14.4833 16.85L9.33333 14.1417L4.18333 16.85L5.16667 11.1167L1 7.05834L6.75833 6.21667L9.33333 1Z"
              stroke="#F25D27"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </li>
        <li
          style="cursor: pointer; position: relative; margin-right: 16px;"
          title="3 star"
        >
          <svg
            fill="#F25D27"
            height="18"
            viewBox="0 0 19 18"
            width="19"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.33333 1L11.9083 6.21667L17.6667 7.05834L13.5 11.1167L14.4833 16.85L9.33333 14.1417L4.18333 16.85L5.16667 11.1167L1 7.05834L6.75833 6.21667L9.33333 1Z"
              stroke="#F25D27"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </li>
        <li
          style="cursor: pointer; position: relative; margin-right: 16px;"
          title="4 star"
        >
          <svg
            fill="#F25D27"
            height="18"
            viewBox="0 0 19 18"
            width="19"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.33333 1L11.9083 6.21667L17.6667 7.05834L13.5 11.1167L14.4833 16.85L9.33333 14.1417L4.18333 16.85L5.16667 11.1167L1 7.05834L6.75833 6.21667L9.33333 1Z"
              stroke="#F25D27"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </li>
        <li
          style="cursor: pointer; position: relative; margin-right: 0px;"
          title="5 star"
        >
          <svg
            fill="none"
            height="18"
            viewBox="0 0 19 18"
            width="19"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.33333 1L11.9083 6.21667L17.6667 7.05834L13.5 11.1167L14.4833 16.85L9.33333 14.1417L4.18333 16.85L5.16667 11.1167L1 7.05834L6.75833 6.21667L9.33333 1Z"
              stroke="#F25D27"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </li>
      </ul>
    </div>
    <p
      class="c4"
    >
      Todos os produtos comercializados s√£o de excelente qualidade, recomendo!
    </p>
  </div>
</div>
`
